src/core/ContactUser.cpp

```diff
@@ -327,6 +327,8 @@ void ContactUser::deleteContact()
     if (m_contactRequest) {
         qDebug() << "Cancelling request associated with contact to be deleted";
         m_contactRequest->cancel();
+    }
+    if (m_contactRequest) {
         m_contactRequest->deleteLater();
```

Comment:
just delete the two new lines and "m_contactRequest->deleteLater();" from here, since it will be called via cancel() anyways.